datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -137,6 +141,10 @@ pub struct PhysicalGroupBy {
     /// expression in null_expr. If `groups[i][j]` is true, then the
     /// j-th expression in the i-th group is NULL, otherwise it is `expr[j]`.
     groups: Vec<Vec<bool>>,
+    // The number of internal expressions that are used to implement grouping
+    // sets. These output are removed from the final output and not in `expr`
+    // as they are generated based on the value in `groups`
+    num_internal_exprs: usize,
```

Comment:
Is there a scenario in which this is something other than `0` or `1`?

Reply:
Currently not. My thinking was that in the future we might add more internal exprs to also solve (https://github.com/apache/datafusion/issues/5672)  and or support more than 64 grouping columns (if someone ever needs that). As I think was done in https://github.com/apache/datafusion/pull/5749